docs/plugins/discovery-file.asciidoc

```diff
@@ -0,0 +1,91 @@
+[[discovery-file]]
+=== File-Based Discovery Plugin
+
+The file-based discovery plugin uses a list of hosts/ports in a `unicast_hosts.txt` file 
+in the `config/discovery-file` directory for unicast discovery.
+
+[[discovery-file-install]]
+[float]
+==== Installation
+
+This plugin can be installed using the plugin manager:
+
+[source,sh]
+----------------------------------------------------------------
+sudo bin/elasticsearch-plugin install discovery-file
+----------------------------------------------------------------
+
+The plugin must be installed on every node in the cluster, and each node must
+be restarted after installation. Note that installing the plugin will add a
+`discovery-file` directory to the `config` folder, and a default `unicast_hosts.txt`
+file that must be edited with the correct unicast hosts list before starting the node.
+
+[[discovery-file-remove]]
+[float]
+==== Removal
+
+The plugin can be removed with the following command:
+
+[source,sh]
+----------------------------------------------------------------
+sudo bin/elasticsearch-plugin remove discovery-file
+----------------------------------------------------------------
+
+The node must be stopped before removing the plugin.
+
+[[discovery-file-usage]]
+[float]
+==== Using the file-based discovery plugin
+
+The file-based discovery plugin provides the ability to specify the 
+unicast hosts list through a simple `unicast_hosts.txt` file that can
+be dynamically updated at any time.  The discovery type for this plugin
+is still the default `zen` plugin, so no changes are required to the 
+`elasticsearch.yml` config file.  This plugin simply provides a facility
+to supply the unicast hosts list for zen discovery through an external
+file that can be updated at any time by a side process.  
+
+For example, this gives a convenient mechanism for Elasticsearch run 
+in docker containers to be dynamically supplied a list of IP addresses
+to connect to for zen discovery when those IP addresses may not be 
+known at node startup.  
+
+Note that the file-based discovery plugin is meant to augment the unicast
+hosts list in `elasticsearch.yml` (if specified), not replace it.  Therefore,
+if there are valid unicast host entries in `discovery.zen.ping.unicast.hosts`,
+they will be used in addition to those supplied in `unicast_hosts.txt`.
+
+Anytime a change is made to the `unicast_hosts.txt` file, even as Elasticsearch
+continues to run, the new changes will be picked up by the plugin and the
+new hosts list will be used for the next pinging round for master election.
+
+Upon installation of the plugin, a default `unicast_hosts.txt` file will
+be found in the `$CONFIG_DIR/discovery-file` directory.  This default file
+will just contain some comments about what the file should contain.  All
+comments should be removed from this file before starting the node and 
```

Comment:
@dakrone I know what you mean. One thing that is good though is that this file *must* be changed before it can be used, the comments are just meant to serve as a placeholder description for what should go in there.  So if they leave the comments in, then they probably didn't change anything and would be alerted to the fact.  Also, these files will actually be typically replaced by machine generated ones.  The alternative would be to use a yaml file or something similar which we decided against.

The suggestion for the comments came from https://github.com/elastic/elasticsearch/pull/20394#issuecomment-246751568